src/odemis/driver/technolution.py

```diff
@@ -160,26 +189,63 @@ def ASM_API_Post_Call(self, url, expected_status, data=None, raw_response=False,
         """
         Call to the ASM API to post data to the ASM API
 
-        :param url (str): url of the command, server part is defined in object variable self._server_url
-        :param expected_status (int): expected feedback of server for a positive call
+        :param url (str): url of the command, server part is defined in object variable self._host
+        :param expected_status (int): expected feedback of server for a successful call
         :param data: data (request body) added to the call to the ASM server (mega field metadata, scan location etc.)
-        :param raw_response (bool): specified the format of the structure returned
+        :param raw_response (bool): Specifies the format of the structure returned. For not raw (False) the content
+        of the response is translated from json and returned. Otherwise the entire response is returned.
         :param timeout (int): [s] if within this period no bytes are received an timeout exception is raised
         :return: status_code(int) or entire response (raw_response=True)
         """
-        logging.debug("Executing: %s" % url)
-        resp = self._session.post(self._server_url + url, json=data, timeout=timeout, **kwargs)
+        logging.debug("Executing POST: %s" % url)
+        resp = self._session.post(self._host + url, json=data, timeout=timeout, **kwargs)
 
         if resp.status_code != expected_status:
-            raise AsmApiException(url, resp, expected_status + 1, self.ASM_API_Get_Call)
-
-        logging.debug("Call to %s went fine, no problems occurred\n" % url)
+            if isinstance(data, dict):
+                for key, value in data.items():
+                    if isinstance(value, list):
+                        try:
+                            # Limit to output of first 10 values to not overload error output message
+                            data[key] = "First 10 values of the list:" + str(value[0:10])
+                        except:
+                            data[key] = "Empty - because data cannot be converted to a string"
+            logging.error("Data dictionary used to make call %s holds the keys:\n %s" % (url, str(data)))
+            raise AsmApiException(url, resp, expected_status, self)
+
+        logging.debug("Call to %s was successful.\n" % url)
         if raw_response:
             return resp
         else:
             return resp.status_code
 
-    def _startCalibration(self, *args):
+    def checkMegaFieldExists(self, mega_field_id, storage_dir):
+        """
+        Check if filename complies with set allowed characters
+        :param mega_field_id (string): name of the mega field.
+        :param storage_dir (string): path to the mega field.
+        :return (bool): True if mega field exists.
+        """
+        ASM_FILE_ILLEGAL_CHARS = r'[^a-z0-9_()-]'
+        ASM_PATH_ILLEGAL_CHARS = r'[^A-Za-z0-9/_()-]'
+        if not re.search(ASM_PATH_ILLEGAL_CHARS, storage_dir):
```

Comment:
It shouldn't have a `not` here (and in the next `if`).

Reply:
True, a test case should have cached this, I implemented also this testcase